project/.gitlab-ci.yml

```diff
@@ -0,0 +1,271 @@
+    ENVIRONMENT: Production
+    STAGE: prod
+    AWS_ACCOUNT_ID: "789012345678" # REPLACEME
+    AWS_ROLE_ARN: arn:aws:iam::${AWS_ACCOUNT_ID}:role/${FILMDROP_DEPLOY_ROLE_NAME}
+    TERRAFORM_STATEFILE: filmdrop-${PROJECT_NAME}-${STAGE}
+    CONSOLE_URL: https://console.${PROJECT_NAME}.stingray.example.com
+
+.setup:
+  script:
+    - echo "Setting AWS credentials on ${STAGE} Account"
+    - /usr/local/bin/aws sts get-caller-identity
+    - /usr/local/bin/aws sts assume-role --role-arn "$AWS_ROLE_ARN" --role-session-name sts > /tmp/$$.sts
+    - export AWS_ACCESS_KEY_ID=`cat /tmp/$$.sts | jq -c '.Credentials.AccessKeyId' | tr -d '"' | tr -d ' '`
+    - export AWS_SECRET_ACCESS_KEY=`cat /tmp/$$.sts | jq -c '.Credentials.SecretAccessKey' | tr -d '"' | tr -d ' '`
+    - export AWS_SESSION_TOKEN=`cat /tmp/$$.sts | jq -c '.Credentials.SessionToken' | tr -d '"' | tr -d ' '`
+    - echo "Creating terraform backend file ..."
+    - echo 'terraform {' >> ${STAGE}.s3.backend.tf
+    - echo '  backend "s3" {' >> ${STAGE}.s3.backend.tf
+    - echo '    encrypt = true' >> ${STAGE}.s3.backend.tf
+    - echo "    bucket = \"filmdrop-${PROJECT_NAME}-${STAGE}-${AWS_REGION}-terraform-state\"" >> ${STAGE}.s3.backend.tf
+    - echo "    dynamodb_table = \"filmdrop-${PROJECT_NAME}-${STAGE}-${AWS_REGION}-state-locks\"" >> ${STAGE}.s3.backend.tf
+    - echo "    key = \"${PROJECT_NAME}-${STAGE}.tfstate\"" >> ${STAGE}.s3.backend.tf
+    - echo "    region = \"${AWS_REGION}\"" >> ${STAGE}.s3.backend.tf
+    - echo '  }' >> ${STAGE}.s3.backend.tf
+    - echo '}' >> ${STAGE}.s3.backend.tf
+
+    - echo "Using FilmDrop Terraform ${FILMDROP_TERRAFORM_RELEASE} release..."
+    - wget -q -O filmdrop-aws-tf-modules-${FILMDROP_TERRAFORM_RELEASE:1}.tgz https://github.com/Element84/filmdrop-aws-tf-modules/archive/refs/tags/${FILMDROP_TERRAFORM_RELEASE}.tar.gz
+    - tar -xzf filmdrop-aws-tf-modules-${FILMDROP_TERRAFORM_RELEASE:1}.tgz
+    - rm filmdrop-aws-tf-modules-${FILMDROP_TERRAFORM_RELEASE:1}.tgz
+    - mv filmdrop-aws-tf-modules-${FILMDROP_TERRAFORM_RELEASE:1}/* .
+    - echo "Building stac-server..."
+    - export FILMDROP_BUILD_DIR=`pwd`
+    - export STAC_SERVER_DIR="stac-server-${STAC_SERVER_TAG:1}"
+    - source $HOME/.nvm/nvm.sh
+    - nvm use v18
+    - curl -L -f --no-progress-meter -o - "https://github.com/stac-utils/stac-server/archive/refs/tags/${STAC_SERVER_TAG}.tar.gz" | tar -xz
+    - cd "$STAC_SERVER_DIR"
+    - npm install
+    - BUILD_PRE_HOOK=true npm run build
+    - mkdir -p $FILMDROP_BUILD_DIR/modules/stac-server/lambda/api
+    - cp dist/api/api.zip $FILMDROP_BUILD_DIR/modules/stac-server/lambda/api/
+    - mkdir -p $FILMDROP_BUILD_DIR/modules/stac-server/lambda/ingest
+    - cp dist/ingest/ingest.zip $FILMDROP_BUILD_DIR/modules/stac-server/lambda/ingest/
+    - mkdir -p $FILMDROP_BUILD_DIR/modules/stac-server/lambda/pre-hook
+    - cp dist/pre-hook/pre-hook.zip $FILMDROP_BUILD_DIR/modules/stac-server/lambda/pre-hook/
```

Comment:
Aren't the stac-server lambda zips included in the terraform module, and thus this build step is just duplication? I'd argue we shouldn't be including the zips in the terraform module _source_, but also think we should have a build step that does pull these down into a build artifact of the whole module so users don't have to worry about any of this. Especially as we add more lambdas to build, i.e., cirrus.